CMakeLists.txt

```diff
@@ -27,7 +28,7 @@ add_library(sparta INTERFACE)
 
 target_include_directories(sparta INTERFACE
       $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
-      $<INSTALL_INTERFACE:include>
+      $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/sparta>
```

Comment:
Are you sure about this?
I tried locally and this doesn't install correctly:
```
$ mkdir build
$ cd build
$ cmake -DCMAKE_INSTALL_PREFIX="../install" -DBUILD_TESTING=OFF ..
$ make
$ make install
Install the project...
-- Install configuration: "Release"
-- Installing: /var/folders/28/bsnpkw693dj4g4r70vb7zn8m0000gn/T/tmp.r5N68DnBSw/SPARTA/install/include/sparta
-- Installing: /var/folders/28/bsnpkw693dj4g4r70vb7zn8m0000gn/T/tmp.r5N68DnBSw/SPARTA/install/include/sparta/sparta
-- Installing: /var/folders/28/bsnpkw693dj4g4r70vb7zn8m0000gn/T/tmp.r5N68DnBSw/SPARTA/install/include/sparta/sparta/HashedAbstractEnvironment.h
-- Installing: /var/folders/28/bsnpkw693dj4g4r70vb7zn8m0000gn/T/tmp.r5N68DnBSw/SPARTA/install/include/sparta/sparta/PatriciaTreeMapAbstractPartition.h
[...]
```
As you see, it installs under include/sparta/sparta, this is wrong.

Reply:
This patch should only change the include directory attached to the generated `sparta_targets.cmake`. If the installed location is wrong, that is unrelated to this PR.

At least when I try it, the current behavior on `main` branch is consistent with what you are seeing:
```
$ cmake -B build -GNinja
$ cmake --install build --prefix $(mktemp -d)
-- Install configuration: "Release"
-- Installing: /tmp/nix-shell.4NHxoh/tmp.8Eye3UvBMT/include/sparta
-- Installing: /tmp/nix-shell.4NHxoh/tmp.8Eye3UvBMT/include/sparta/sparta
-- Installing: /tmp/nix-shell.4NHxoh/tmp.8Eye3UvBMT/include/sparta/sparta/AbstractDomain.h
-- Installing: /tmp/nix-shell.4NHxoh/tmp.8Eye3UvBMT/include/sparta/sparta/PatriciaTreeKeyTrait.h
-- Installing: /tmp/nix-shell.4NHxoh/tmp.8Eye3UvBMT/include/sparta/sparta/LiftedDomain.h
[...]
-- Installing: /tmp/nix-shell.4NHxoh/tmp.8Eye3UvBMT/cmake/sparta_target.cmake
```
If you then inspect the contents of `sparta_target.cmake`, you'll see:
```cmake
get_filename_component(_IMPORT_PREFIX "${CMAKE_CURRENT_LIST_FILE}" PATH)
get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
if(_IMPORT_PREFIX STREQUAL "/")
  set(_IMPORT_PREFIX "")
endif()

# Create imported target sparta
add_library(sparta INTERFACE IMPORTED)

set_target_properties(sparta PROPERTIES
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include"
  INTERFACE_LINK_LIBRARIES "Boost::thread"
)
```
So the `INTERFACE_INCLUDE_DIRECTORIES` would point to `/tmp/nix-shell.4NHxoh/tmp.8Eye3UvBMT/include`, which wouldn't be right (it should point to `include/sparta` instead).

I confirmed this by generating a dummy CMake project:
```
$ cd $(mktemp -d)
$ cat > CMakeLists.txt <<EOF
cmake_minimum_required(VERSION 3.20)
project(Example)
find_package(Boost REQUIRED COMPONENTS thread)
include("/tmp/nix-shell.4NHxoh/tmp.8Eye3UvBMT/cmake/sparta_target.cmake")
add_executable(example example.cpp)
target_link_libraries(example PRIVATE sparta)
EOF

$ cat > example.cpp <<EOF
#include <sparta/PowersetAbstractDomain.h>
int main() { return 0; }
EOF

$ cmake -B build -GNinja
$ cmake --build build
/nix/store/90h6k8ylkgn81k10190v5c9ldyjpzgl9-gcc-wrapper-12.3.0/bin/g++ -DBOOST_THREAD_DYN_LINK -DBOOST_THREAD_NO_LIB -isystem /tmp/nix-shell.4NHxoh/tmp.8Eye3UvBMT/include  -MD -MT CMakeFiles/example.dir/example.cpp.o -MF CMakeFiles/example.dir/example.cpp.o.d -o CMakeFiles/example.dir/example.cpp.o -c /tmp/nix-shell.4NHxoh/tmp.43SeFBEwQS/example.cpp
/tmp/nix-shell.4NHxoh/tmp.43SeFBEwQS/example.cpp:1:10: fatal error: sparta/PowersetAbstractDomain.h: No such file or directory
    1 | #include <sparta/PowersetAbstractDomain.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
ninja: build stopped: subcommand failed.
```

With this patch, the dummy project builds successfully instead.

Reply:
I see. I do think we want to change this and force users to `include <sparta/XXX>` instead of `<XXX>` directly.
Let me try to make that change internally, then you can rebase on top of it?

Reply:
Thanks. I've updated the include directories to be consistent with the new install location, and I've modified the commit message accordingly.